benchmark/contractions.jl

```diff
@@ -0,0 +1,75 @@
+function extract_contract_labels(contraction::AbstractString)
+  symbolsC = match(r"C\[([^\]]*)\]", contraction)
+  labelsC = split(symbolsC.captures[1], ","; keepempty=false)
+  symbolsA = match(r"A\[([^\]]*)\]", contraction)
+  labelsA = split(symbolsA.captures[1], ","; keepempty=false)
+  symbolsB = match(r"B\[([^\]]*)\]", contraction)
+  labelsB = split(symbolsB.captures[1], ","; keepempty=false)
+  return labelsC, labelsA, labelsB
+end
+
+function generate_contract_benchmark(
+  line::AbstractString; T=Float64, alg=default_contract_alg(), do_alpha=true, do_beta=true
```

Comment:
```suggestion
  line::AbstractString; eltype=Float64, alg=default_contract_alg(), do_alpha=true, do_beta=true
```
More descriptive, and also anticipates that we might want to benchmark other tensor types as well.

Reply:
`eltype` might also not be great though, given that it is the name of the function `Base.eltype`, which you now have to remember that you cannot call anymore. We can do `elt`? (also, this really is an internal function that does not really work with anything but arrays and is designed specifically for parsing that file, so I'm not sure it matters?)

Reply:
Yes, not the biggest deal, but still I like to keep some consistency of naming across libraries.

I often go back and forth on that issue of calling arguments `eltype` vs. `elt`. You can still call the function as `Base.eltype` but that isn't ideal. `elt` is fine with me.

Reply:
Agreed, I guess we haven't really discussed naming conventions, but generally I have used `T` when it is a type parameter, but then `elt` when it is an argument.